### PR TITLE
Update golangci-lint to 1.21

### PR DESCRIPTION
**Pull request by howardjohn**
WIP this won't work without some prow change and a proper common files update

### CODE REVIEW

**Review by geeknoid — APPROVED** [review]

File: galley/pkg/config/analysis/analyzers/injection/injection-version.go [hunk 1]
```diff
@@ -137,12 +137,11 @@ func tryReturnSidecarInjectorVersion(p *v1.Pod) string {
 // getContainerNameVersion parses the name and version from a container image.
 // If the version is not specified or can't be found, version is the empty
 // string.
-func getContainerNameVersion(c *v1.Container) (image string, version string) {
+func getContainerNameVersion(c *v1.Container) (version string) {
```
Comment (hzxuzhonghu) [inline galley/pkg/config/analysis/analyzers/injection/injection-version.go]: getContainerImageVersion

**Comment by hzxuzhonghu** [comment]
Why so many licenses? Are they added by hand?

**Comment by howardjohn** [comment]
No they are from `make generate`, there is some tool that extracts them
all. Apparently some new legal requirement..

**Review by esnible — APPROVED** [review]
This review is for istioctl/... only.  I did not review the rest.

**Comment by hzxuzhonghu** [comment]
>No they are from `make generate`, there is some tool that extracts them
all. Apparently some new legal requirement..

Good, it is better if doc it explicitly for developers.

**Pull request merged**